File: ELEC881_Copilot/src/No Description Prompt/Java script (JS)/q10-non-overlapping-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
    intervals.sort((a, b) => a[1] - b[1]);
    let count = 0;
    let prev = intervals[0];
    for (let i = 1; i < intervals.length; i++) {
        if (intervals[i][0] < prev[1]) {
            count++;
        } else {
            prev = intervals[i];
        }
    }
    return count;
    
};

File: ELEC881_Tabnine/src/No Description Prompt/Python/q10-non-overlapping-intervals.py
class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        if not intervals:
            return 0

        intervals.sort(key=lambda x: x[1])
        count, end = 1, intervals[0][1]

        for i in range(1, len(intervals)):
            if intervals[i][0] >= end:
                count += 1
                end = intervals[i][1]
            else:
                end = min(end, intervals[i][1])

        return len(intervals) - count
        